### PR TITLE
AEA-1471 CI plugins install check

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

CI plugins install check

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -207,7 +207,34 @@ jobs:
       - name: Check aea dependenices and imports
         run: |
           tox -e dependencies_check
-
+  plugins_install_check:
+    continue-on-error: False
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:214]: windows?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -207,7 +207,34 @@ jobs:
...
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@master
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Install tox
+        run : |
+          pip install tox
+      - name: Check plugin aea-ledger-cosmos
+        run : |
+          tox -r -e plugins_env -- /bin/sh -c "pip install ./plugins/aea-ledger-cosmos && aea generate-key cosmos && echo aea-ledger-cosmos checked!"
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:228]: That won't work. You need to install the local version. Not the one from pypi. (see how its done in tox.ini)

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (solarw) [reply .github/workflows/workflow.yml]: fixed

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (solarw) [reply .github/workflows/workflow.yml]: it's done in tox.ini as a dependency: `deps = .[all]`

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -207,7 +207,39 @@ jobs:
...
+          tox -r -e plugins_env -- sh -c "pip install ./plugins/aea-ledger-cosmos && aea generate-key cosmos && echo aea-ledger-cosmos checked!"
+      - name: Check plugin aea-ledger-ethereum
+        run : |
+          tox -r -e plugins_env -- sh -c "pip install ./plugins/aea-ledger-ethereum && aea generate-key ethereum && echo aea-ledger-ethereum checked!"
+      - name: Check plugin aea-ledger-fetchai
+        run : |
+          tox -r -e plugins_env -- sh -c "pip install ./plugins/aea-ledger-fetchai && aea generate-key fetchai && echo aea-ledger-fetchai checked!"
+      - name: Check plugin aea-cli-ipfs
+        run : |
+          tox -r -e plugins_env -- sh -c "pip install ./plugins/aea-cli-ipfs && aea ipfs --help && echo aea-cli-ipfs checked!"
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:242]: Are we sure the `pip install` commands will not install the `aea` on PyPI?

**Review by marcofavorito — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**